### ISSUE TITLE
nvpy is just not working anymore

### ISSUE THREAD

**Issue by JENKOP** [cpbotha/nvpy#238]
Sync issues. Python 3 problem. SInce I installed nvpy with apt I removed it and installed it with pip

Still cant sync. 

Sync step 1 error. Could not update note.
some hex numbers....
server{'update_error':<HTTPError 404: 'Not Found'>, 'get_error':

**Comment by yuuki0xff** [comment]
Please tell me the following info:

- nvPY version
- OS and Python version
- Complete logs since nvPY start. Log files are exist on `~/.nvpy/nvpy.log`.

**Comment by JENKOP** [comment]
1. nvPY 2.3.1

2a. Bodhi Linux, basically 
Distributor ID:Ubuntu
Description:	Ubuntu 22.04 LTS
Release:	22.04
Codename:	jammy

2b. Python 3.10.12

3.Log is full of this and only this (example):

2024-05-01 17:02:28,461 - ERROR - Could not get/update note af9a8866398e4e07b0fec413956362d0: update_error=HTTP Error 404: Not Found, get_error=HTTP Error 404: Not Found
2024-05-01 17:02:28,462 - DEBUG - Updating note 8dc6e0c3-8e25-419f-9f65-34398a3ce90c (local key 8dc6e0c3-8e25-419f-9f65-34398a3ce90c) to server.
2024-05-01 17:02:33,357 - ERROR - Could not get/update note 8dc6e0c3-8e25-419f-9f65-34398a3ce90c: update_error=HTTP Error 404: Not Found, get_error=HTTP Error 404: Not Found

**Comment by yuuki0xff** [comment]
It's strange that the updating note causes 404 errors.

Did you delete notes from other clients while starting nvpy?
Does the problem stay even after a full synchronization? 
Is only two notes (8dc6e0c3-8e25-419f-9f65-34398a3ce90c, af9a8866398e4e07b0fec413956362d0) having synchronizing issue?

**Comment by JENKOP** [comment]
No. I think it is basically all notes. Possibly a hundred, When I experienced problems, I opened new simplenote account. After a failed sync attempt, I tried to delete notes directly in simplenote.com

**Comment by yuuki0xff** [comment]
If you are okay with discarding remote notes, you can upload all notes from local to remote using the nvpy-db-utils.py.
https://github.com/cpbotha/nvpy/blob/master/debug-utils/nvpy-db-utils.py

```bash
# Back up the nvpy database.
cp -a ~/.nvpy/ ~/.nvpy.backup

# Try to upload a note.
./debug-utils/nvpy-db-utils.py delete --remote af9a8866398e4e07b0fec413956362d0
./debug-utils/nvpy-db-utils.py update --direction upload af9a8866398e4e07b0fec413956362d0

# Try to upload all notes.
./debug-utils/nvpy-db-utils.py delete --remote --all
./debug-utils/nvpy-db-utils.py update --direction upload --all
```

UPDATE: 2024-12-10 Change commit ID to branch name.

**Comment by yuuki0xff** [comment]
I fixed the nvpy-db-utils.py crash issue. Available on the master branch.

>  When I experienced problems, I opened new simplenote account.

The first issue remains unclear. If you have logs when this issue first occurred, please share me.

**Comment by jvmiles** [comment]
"I fixed the nvpy-db-utils.py crash issue. Available on the master branch."

Should I update it? How? pip install?

**Comment by yuuki0xff** [comment]
Download from https://github.com/cpbotha/nvpy/blob/master/debug-utils/nvpy-db-utils.py and run it.
If you installed nvpy with pip, the required modules (`nvpy` and `simplenote`) have already been installed.

**Comment by jvmiles** [comment]
Will this shit ever work again?
On one computer I have synchronization issues, on the other one it crashes without error. This software has become a nightmare after serving me many years. If things can't be fixed, is there another good simplenote client for Linux that actually works?

**Comment by jvmiles** [comment]
I deleted a note that made problems. No difference. It synchonizes into oblivion. I deleted all notes on simplenot for a fresh synronization. nvpy error log is full of

2024-12-08 16:43:44,455 - DEBUG - Updating note af9a8866398e4e07b0fec413956362d0 (local key af9a8866398e4e07b0fec413956362d0) to server.
2024-12-08 16:43:44,908 - ERROR - Could not get/update note af9a8866398e4e07b0fec413956362d0: update_error=HTTP Error 404: Not Found, get_error=HTTP Error 404: Not Found

**Comment by yuuki0xff** [comment]
Could you please provide the steps you took, the logs, and the results in chronological order? Without being able to reproduce the issue, it will be difficult to investigate and fix

**Comment by yuuki0xff** [comment]
> is there another good simplenote client for Linux that actually works?

Linux version of Simplenote (built with Electron)

https://github.com/Automattic/simplenote-electron/releases

Using app.simplenote.com through a web browser is another option.

**Comment by yuuki0xff** [comment]
> When I experienced problems,

Regarding the issue that initiated this matter, investigation is impossible due to a lack of information.

> I opened new simplenote account. After a failed sync attempt, I tried to delete notes directly in simplenote.com
> https://github.com/cpbotha/nvpy/issues/238#issuecomment-2089181102

> I created a new simplenode account and run the nvpy as another user (copied files) and C&P into Simplenode.
> https://github.com/cpbotha/nvpy/issues/240#issuecomment-2530277820

I seems to you might be making the problem worse yourself.

nvPY's sync system performs bidirectional synchronization between the local database and Simplenote. To achieve this, it keeps track of the sync status in the local database. The following actions are not supported and could lead to issues:

- Switching to a different Simplenote account without clearing the contents of ~/.nvpy/.
- Copying ~/.nvpy/ to another computer and using it as-is.

These actions could cause inconsistencies in the local database. This might be the root cause of the various problems you're experiencing in your environment. If you store valuable notes in Simplenote, you should avoid reckless attempts to fix issues.

If you intend to continue using nvPY, you should follow these steps:

1. Stop nvPY and back up `~/.nvpy`.
2. Consolidate into a single Simplenote account. Migrate your data in Simplenote and update your `.nvpy.cfg` file accordingly.
3. If there are notes in nvPY's local database that are not reflected in Simplenote, manually transfer them to Simplenote.
4. Ensure all notes are latest on simplenote.
5. Delete the `~/.nvpy` directory.
6. Restart nvPY.

**Comment by jvmiles** [comment]
The problem is solved.
I C&P all the stuff online per hand into another simple note account. Worked, but does not inspire confidence since I rely on this software a lot.

**Comment by ernstki** [comment]
> The problem is solved. I C&P all the stuff online per hand into another simple note account. Worked, but does not inspire confidence since I rely on this software a lot.

It's fine to be pissed when something isn't working right, but making crass remarks and dumping negativity on the maintainer(s) in the issue tracker is not going to achieve your aims, unless your aim is to burn out the maintainers. I'd rather you didn't, since I use nvpy, too.

It was already suggested that you switch to [the official Simplenote desktop client](https://simplenote.com/#available-on-all-your-devices) if you're unhappy with nvpy. It's _also_ maintained by basically volunteers within Automattic, but because they run the whole service and can look at backend logs and stuff, you're likely to have fewer sync problems… at the expense of having a bulky Electron app running all the time.
